### PR TITLE
feat: add support for agent configuration

### PR DESCRIPTION
**Pull request by dmurray-lacework**
## Summary

Add support for agent configuration https://docs.lacework.net/onboarding/restricted/configure-agent-behavior-in-configjson-file

Add new variable `lacework_agent_configuration`

## How did you test this change?

- [x] TF plan/TF Apply `examples/additional-agent-configuration`

## Issue

https://lacework.atlassian.net/browse/GROW-2391

Closes https://github.com/lacework/terraform-aws-ssm-agent/issues/58

### CODE REVIEW

File: examples/additional-agent-configuration/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  lacework_agent_tags = {
+    env = "dev"
+  }
+
+  aws_resources_tags = {
+    billing = "testing"
+    owner   = "myself"
+  }
+
+    lacework_agent_configuration = {
```
Comment (afiune) [inline examples/additional-agent-configuration/README.md:24]: indentation

File: examples/additional-agent-configuration/main.tf [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+module "lacework_aws_ssm_agents_install" {
+  source = "../../"
+
+  lacework_agent_tags = {
+    env = "dev"
+  }
+
+  lacework_enable_default_syscall_config = "true"
```
Comment (afiune) [inline examples/additional-agent-configuration/main.tf:12]: This is not in the README 🤔 - So update either of them to match.

File: examples/additional-agent-configuration/main.tf
Comment (afiune) [inline examples/additional-agent-configuration/main.tf:1]: You need to add this new example to the `scripts/ci_tests.sh` file

**Review by afiune — CHANGES_REQUESTED** [review]

**Review by JPLachance — APPROVED** [review]
LGTM!

**Review by afiune — APPROVED** [review]

File: setup_lacework_agent.sh [hunk 3]
```diff
@@ -62,6 +64,11 @@ render_agent_config() {
     _server_url_json='"serverurl": "'$SERVER_URL'",'
   fi
 
+  # Additional Config Json
+  if [ "$ADDITIONAL_CONFIG" != "" ]; then
+    _additional_config_json='$ADDITIONAL_CONFIG,'
```
Comment (JPLachance) [inline setup_lacework_agent.sh:69]: I just found this in our logs:
```
time="2023-11-02T20:54:57.451Z" level=error msg="Cannot parse config file [/var/lib/lacework/config/config.json] unable to parse content: [invalid character '$' looking for beginning of object key string]" caller="json.go:325" pid=27849
```

The cause:
```
root@ip-10-6-10-92:/var/log/lacework# cat /var/lib/lacework/config/config.json
{
  "tokens": { "AccessToken": "Nope" },

  $ADDITIONAL_CONFIG,
  "tags": {"account_id":"none","env":"dev","region":"us-east-1"}
}
```

Basically, it looks like we are rendering `$ADDITIONAL_CONFIG,` in the file, which is invalid.

Any idea?

Comment (dmurray-lacework) [reply setup_lacework_agent.sh:69]: Hey @JPLachance taking a look at this now

Comment (dmurray-lacework) [reply setup_lacework_agent.sh:69]: Fix PR -> https://github.com/lacework/terraform-aws-ssm-agent/pull/63

**Pull request merged**